go/tools/gazelle/packages/walk.go

```diff
@@ -28,6 +29,14 @@ import (
 	"strings"
 )
 
+// packageReader holds the state used to read package metadata from a directory.
+// This mostly exists to avoid passing the same of parameters to every function.
+type packageReader struct {
+	bctx               build.Context
+	repoRoot, goPrefix string
```

Comment:
Done. Originally, I just had one `packageReader`, and I set the `dir` each time the callback passed to `filepath.Walk` was called.